src/util/mediatypes.js

```
@@ -0,0 +1,117 @@
+
+      type = data.type = type || this.checkUrl( baseUrl );
+      callback = callback || function(){};
+
+      if ( type === "youtube" ) {
+        parsedUri = URI.parse( baseUrl );
+        id = parsedUri.queryKey.v || parsedUri.directory.replace( "/", "" );
+        if ( !id ) {
+          return;
+        }
+
+        xhrURL = "https://gdata.youtube.com/feeds/api/videos/" + id + "?v=2&alt=jsonc&callback=?";
+        Popcorn.getJSONP( xhrURL, function( resp ) {
+          var raw = resp.data;
+          if ( !raw ) {
+            return;
+          }
+          data.source = "http://www.youtube.com/watch?v=" + id;
+          data.title = raw.title;
+          data.thumbnail = raw.thumbnail.hqDefault;
+          data.author = raw.uploader;
+          data.duration = raw.duration;
+          data.denied = raw.accessControl.embed === "denied";
+          callback( data );
+        });
+      } else if ( type === "soundcloud" ) {
+        parsedUri = URI.parse( baseUrl );
+        splitUriDirectory = parsedUri.directory.split( "/" );
+        id = splitUriDirectory[ splitUriDirectory.length - 1 ];
+        xhrURL = "http://api.soundcloud.com/tracks/" + id + ".json?callback=?&client_id=PRaNFlda6Bhf5utPjUsptg";
+        Popcorn.getJSONP( xhrURL, function( raw ) {
+          if ( !raw ) {
+            return;
+          }
+
+          if ( raw.sharing === "private" || raw.embeddable_by === "none" ) {
+            data.denied = true;
+          }
+          data.source = baseUrl;
+          data.thumbnail = raw.artwork_url;
+          data.duration = raw.duration / 1000;
+          data.title = raw.title;
+          data.hidden = true;
+          callback( data );
+        });
+      } else if ( type === "vimeo" ) {
+        parsedUri = URI.parse( baseUrl );
+        splitUriDirectory = parsedUri.directory.split( "/" );
+        id = splitUriDirectory[ splitUriDirectory.length - 1 ];
+        xhrURL = "http://vimeo.com/api/v2/video/" + id + ".json?callback=?";
```

Same deal, don't hide error cases.